Add Horizontal and Vertical Content Alignment Options to ScrollContainer

KnifeXRage opened the pull request:
Closes: [#13421](https://github.com/godotengine/godot-proposals/issues/13421)

Added two new properties to ScrollContainer:
- `horizontal_content_align`
- `vertical_content_align`

Allows ScrollContainer content to be aligned left/center/right and top/center/bottom from Inspector or using GDScript.
And also updated Documentation in doc/classes/ScrollContainer.xml.

Here is a demo video of `ScrollContainer` Content Alignment Feature:

[ScrollContainer Alignment Demo](https://github.com/user-attachments/assets/35361508-488c-4591-80c3-0ef54f2daa36)

KoBeWi committed:
Add change_scene_to_node()

KoBeWi committed:
Improve to_string() and add it to Resource

Alex2782 committed:
Fix 'exclusive' child window

HolonProduction committed:
Autocompletion: Use correct completion type for argument options

ryevdokimov committed:
Move editor viewport gizmos visibility logic out of process notification

ryevdokimov committed:
Remove redundant line when toggling cinematic preview

ryevdokimov committed:
Condition editor viewport nav modifier checks on freelook / mouse is captured

KoBeWi committed:
Add switch on hover to TabBar

Meorge committed:
Add Instant Preview to Quick Open dialog

Add toggle for instant preview

Always keep search box selected so that keyboard navigation works

Add default setting for Instant Preview

Directly set property value for resource via Quick Load menu (no undo/redo or dirty-scene functionality yet)

Add undo/redo functionality

Update class reference

Update doc/classes/EditorSettings.xml

Co-authored-by: Micky <66727710+Mickeon@users.noreply.github.com>

Slight improvement(?) to wording of setting

Allow previewing without committing change

Address various suggestions/improvements

Only allow Instant Preview to be used if Quick Open menu is being used to modify a property

Only allow property-based Quick Load when resource to modify is defined (otherwise default to old behavior)

Apply suggestions from code review

Co-authored-by: Tomasz Chabora <kobewi4e@gmail.com>

Address comments/suggestions

Get rid of duplicated code and use original callback strategy

(Attempt to) fix Instant Preview for editing multiple nodes at once and undo/redo stack for single nodes

Fix cancelling Quick Open when multiple nodes are selected

Prevent initially selected item in Quick Open dialog from overwriting the currently selected property

Apply suggestions from code review

Co-authored-by: Tomasz Chabora <kobewi4e@gmail.com>
Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

Make a few changes/improvements based on feedback

- Combine some duplicated code into `_finish_dialog_setup()`
- Move `ERR_FAIL_NULL(p_obj);` to top of checks
- Fix renaming of `is_instant_preview_enabled()` across code, and remove now-redundant conditions where it is used
- Make `EditorResourcePicker::property_path` be `StringName` not `String`

passivestar committed:
Add support for closing dialog windows with Cmd+W on macOS

dsnopek committed:
Allow editor plugins to modify run arguments

DexterFstone committed:
Fix: Tilemap tools not overriding main editor tools

timothyqiu committed:
Cleanup editor translation related methods

- Unify logic for loading editor/property/doc/extractable translations.
- Replace legacy `TranslationServer` methods with translation domains for internal translations.
- Only pre-create editor/property/doc translation domains in editor builds.
- Prevent adding `null` translation.
- Fixes potential loading of duplicated editor translations.
- Add internal `has_translation_for_locale()` instead of calling `get_loaded_translations().has()`.

raulsntos committed:
[.NET] Remove EFS update on reloading assemblies

Updating the files at this point is too early, the scripts haven't been updated yet so their type info is empty.

It looks like EFS is updated anyway without doing it explicitly in `LookupScriptsInAssembly`, and it happens after the scripts have updated their info. So this code seems unnecessary.

raulsntos committed:
[.NET] Fix various errors in bindings generator

- Fix `is_intentionally_ignored` check, the names in the hash map are not fully-qualified.
- Add a special case for `object.ToString`.
- Validate API type before attempting to reference a type member.

aaronfranke committed:
GLTF: Move accessor and buffer view Dictionary conversion into classes

stuartcarnie committed:
Metal: Fix `texture_get_data` other linear formats

Introduce a specialised `texture_get_data` for `RenderDeviceDriver`,
which can retrieve the texture data from the GPU driver for shared
textures (`TEXTURE_USAGE_CPU_READ_BIT`).

Closes #108115

Giganzo committed:
Update Breeze dark theme colors

timothyqiu committed:
Move plural rules logic into a separate class

- Extracts plural rules logic in `TranslationPO` into a new `PluralRules` class.
- Changes caching the last used plural index in `TranslationPO` into an LRU cache in `PluralRules`.
- Adds tests for `PluralRules`.

timothyqiu committed:
Add default plural rules

This makes the PO loader correctly handle the situation where the optional
`Plural-Forms` header field does not exist.

The `Translation` class and its subclasses always have access to valid plural
rules via `_get_plural_rules()`. Plural rules are prioritized:

1. `Translation.plural_rules_override`
2. `TranslationServer.get_plural_rules(locale)`
3. The English plural rules: `nplurals=2; plurals=(n != 1)`

Co-Authored-By: Pāvels Nadtočajevs <7645683+bruvzg@users.noreply.github.com>

timothyqiu committed:
Move context and plural support to Translation

- `TranslationPO` is now an empty class. It exists for compatibility.
- `OptimizedTranslation` stays the same, no context or plural support.

GabCoolDude committed:
View resource signals in the Connections Dock

pippenpaddleopsicopolis committed:
Removes the need to have a display name for built-in action overrides

precup committed:
Remove spurious curve changed signals

dsnopek committed:
Make `OpenXRCompositionLayer` and its children safe for multithreaded rendering

vaner-org committed:
Allow reconnecting AnimationNodeStateMachine transitions

atlasapplications committed:
Add Stretch Modes for Splash Screen

Co-authored-by: Samuel Pedrajas <samuelpedrajaspz@gmail.com>

vaner-org committed:
Correctly resets StateMachine opacity when a node is played or renamed

BlueCube3310 committed:
Betsy: Generate BC1 tables once during compression

dsnopek committed:
OpenXRExtensionWrapper: Document how to safely work when rendering with "Separate" thread model

vaner-org committed:
Make BlendSpace cursor default, and more like StateMachine

vaner-org committed:
Double-click BlendSpace points to open their editor

clayjohn committed:
Use half float precision buffer for 3D when HDR2D is enabled

This is necessary for Environment effects like Glow to work correctly.

BlueCube3310 committed:
Betsy: Convert RGB to RGBA textures on the GPU

vaner-org committed:
Better deselection handling in AnimationTree editors

AayushSarikhada committed:
Fix Android back gesture failing after keyboard dismissal (#109806)

Reason of bug: This happened because the text input field was still receiving the first back gesture event instead of letting it reach the main app.

BlueCube3310 committed:
Image: Optimize format conversion

BlueCube3310 committed:
Image: Fix normalization of mipmaps for half and float formats

devloglogan committed:
Allow custom monitors to select desired type

timothyqiu committed:
Make text-related nodes translation domain aware

- Makes `is_layout_rtl()` translation domain aware
- Makes various text-drawing controls translation domain aware
- Makes translation preview use the project's fallback locale when disabled

blueskythlikesclouds committed:
Overhaul screen space reflections.

arkology committed:
Expand `LineEdit` with metadata name in `AddMetadataDialog`

DallasHuff committed:
Add New File Shortcuts in FileSystem Dock

It currently takes several clicks to create a file in the filesystem dock. This
change adds the option to create shortcuts for creating files,
greatly reducing the number of inputs required to do so.

DeeJayLSP committed:
Add `reserve()` to `Dictionary`, apply to constructors on GDScript VM

arkology committed:
`ProjectManager`: Change favorite button tooltip to toggle

Giganzo committed:
Add setting for when to show the focus state for mouse input

KoBeWi committed:
Reduce repetitive code in FoldableContainer

YeldhamDev committed:
Unfold tree items on hover while drag-n-dropping

Co-authored-by: suddjian

YeldhamDev committed:
Fix "Region Editor" stuck in drag state when closing

Calinou committed:
Show centiseconds in LightmapGI bake time printout in the editor

On modern high-end GPUs, bake times are regularly under 15 seconds,
especially with preview-quality settings.

In this case, sub-second differences can be significant for benchmarking,
so it's worth showing them.

KoBeWi committed:
Fix Script editor state types

Tcarr20 committed:
Fix editor log search ignoring BBCode formatting in messages

When searching in the editor output log, messages with BBCode formatting
(like [color=red]text[/color]) would not be found when searching for the
plain text content. For example, searching for "Example is" in a message
"[color=blue]EXAMPLE[/color] is super cool" would fail to find the match.

Commit:
Adding a vertical splitter to the create node dialog so the descriptions are easier to read

Adding a vertical splitter to the create node dialog so the descriptions are easier to read

clang-format conformity

xuhuisheng committed:
fix snap Polygon2D editor

Alexofp committed:
Option to enable cvtt and betsy compression in exported builds

Commit:
Misc improvements to the `GodotPlugin` API

- Add overload method for `GodotPlugin#emitSignal(...)`
- Allow passing `null` values as signal arguments

WhalesState committed:
Group virtual keyboard properties in the inspector

KoBeWi committed:
Remove side menu functionality in EditorFileDialog

SantiagoDaza3 committed:
LineEdit: now correctly emits text_changed on Unicode code-point and control character insertion.

Commit:
System-provided ENet warning

Warning from documentation: https://github.com/godotengine/godot/tree/master/thirdparty#enet

stuartcarnie committed:
Build: Fix container build path for swift

syntaxerror247 committed:
Android Editor: Add game speed control options in game menu bar

TokageItLab committed:
Make StateMachinePlayback set `Start` state as default in constructor

Alex2782 committed:
Fix custom icon in Android export

clayjohn committed:
Add all PowerVR devices to the ban list for disabling the transform feedback shader cache

aaronfranke committed:
GLTF: Support animating node visibility

TokageItLab committed:
Change Vector to LocalVector in SpringBone

akien-mga committed:
EditorRun: Load `override.cfg` to get window configuration for embedded mode

Kaleb-Reid committed:
Use correct ndc for proximity fade in Compatibility

bruvzg committed:
Add MIME argument to the `FileDialog.add_filter`.

YeldhamDev committed:
Fix some dragging operations in the editor breaking when tabbing out

paul-marechal committed:
Add note for some usages of Plane in APIs

Some old APIs that may expect a Vector4 instead require a Plane: this
is pretty confusing especially with tangents as one could interpret
the API as wanting a value directly representing the tangent plane when
in reality the Plane type is just used to carry one Vector3 and one
extra sign component.

This commit attempts to clear any potential confusion by adding a note
to the APIs using Planes in such a way.

raulsntos committed:
Make `MissingNode`/`MissingResource` non-virtual and hide from dialogs

This allows ClassDB to get the default values for inherited properties.
The `MissingNode` and `MissingResource` types are excluded from the create
dialog and the inspector's Resource selector.

HolonProduction committed:
LSP: Fix goto native declaration

passivestar committed:
Implement orbit snapping in 3D viewport

Joy-less committed:
Fix IsStaticallyResolvable

KnifeXRage committed:
Update .gitignore

KnifeXRage committed:
Update .gitignore

KnifeXRage committed:
Merge branch 'master' of https://github.com/KnifeXRage/godot-fork

aaronfranke committed:
GLTF: Enforce `STEP` interpolation for integer and boolean animations

kitbdev committed:
Fix indented line wrap with no spaces

KoBeWi committed:
Improvements to ProjectManager's _update_theme()

DarioSamo committed:
Organize render surface sorting key for optimizing API performance.

bruvzg committed:
[macOS] Fix ~500ms hang on transparent OpenGL window creation on macOS 26.

brycehutchings committed:
Fix D3D12 driver returning internal types to RenderingDevice's texture_get_native_handle/get_driver_resource

Repiteo committed:
Core: Consolidate typed container logic

LanzaSchneider committed:
fix: compilation errors when `disable_xr=yes`

aaronfranke committed:
Preserve mesh, material, and texture names in GLTF export

OverShifted committed:
Fix typo in mutex documentation

Change "more times that" to "more times than"

KnifeXRage committed:
Added Content Alignment Feature

Add Vertical and Horizontal content Alignment options to ScrollContainer

KnifeXRage committed:
Added Content Alignment Feature

Add Vertical and Horizontal content Alignment options to ScrollContainer

KnifeXRage committed:
Fixed Scroll To Center Bug

KnifeXRage committed:
Fixed Scroll To Center Bug

KnifeXRage committed:
Added Docs

KnifeXRage committed:
Added Docs

AThousandShips commented on scene/gui/scroll_container.h:
```diff
@@ -48,11 +48,26 @@ class ScrollContainer : public Container {
 	};
 
+	enum ContentHAlign {
+		H_ALIGN_LEFT = 0,
```
```suggestion
		H_ALIGN_LEFT,
```
For all, redundant

AThousandShips commented on scene/gui/scroll_container.cpp:
```diff
@@ -360,6 +360,46 @@ void ScrollContainer::_reposition_children() {
+		int align_v = static_cast<int>(vertical_content_align);
+		float extra = viewport_h - total_content_h;
+		if (align_v == V_ALIGN_CENTER) {
+			group_v_offset = floorf(extra * 0.5f);
```
```suggestion
			group_v_offset = Math::floor(extra * 0.5f);
```
For all

KnifeXRage committed:
Update scene/gui/scroll_container.h

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

KnifeXRage committed:
Update scene/gui/scroll_container.h

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

KnifeXRage committed:
Update scene/gui/scroll_container.h

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

KnifeXRage committed:
Update scene/gui/scroll_container.h

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

AThousandShips commented:
Thanks for opening a pull request!

Feature pull requests should be associated to a feature proposal to justify the need for implementing the feature in Godot core. Please open a proposal on the [Godot proposals](https://github.com/godotengine/godot-proposals) repository (and link to this pull request in the proposal's body).

KoBeWi commented:
How is this different from using size flags?

https://github.com/user-attachments/assets/1a987fe7-635b-4b77-a1b3-6e7b22b31006

AThousandShips commented on scene/gui/scroll_container.cpp:
```diff
@@ -824,3 +909,29 @@ ScrollContainer::ScrollContainer() {
 }
+
+void ScrollContainer::set_horizontal_content_align(int p_align) {
+	int clamped = CLAMP(p_align, 0, 2);
```
This shouldn't use clamp, it should use a range check instead, also those should be in the correct order in the file, above where `set_h_scroll` is declared

KoBeWi commented:
But it doesn't need any "extra" nodes. The ScrollContainer already requires one content node (BoxContainer, GridContainer etc.), and that node is the subject to size flags. You even have one in your video, I replicated that layout. You can set the alignment from GDScript using `set_anchors_and_offsets_preset()`.

KnifeXRage committed:
Merge branch 'godotengine:master' into content_alignment_scroll_container

ultsi committed:
LSP: Fix dict[] uses where the key might not exist

adamscott committed:
[macOS] Fix microphone issue

Repiteo committed:
Merge pull request #111652 from DarioSamo/opaque-list-key

Organize render surface sorting key for optimizing API performance.

Repiteo committed:
Merge pull request #108422 from stuartcarnie/108115_metal_cpu_read_bit

Metal: Fix `texture_get_data` other linear formats

Repiteo committed:
Merge pull request #111657 from bruvzg/mac26_gl_hang

[macOS] Fix ~500ms hang on transparent OpenGL window creation on macOS 26.

Repiteo committed:
Merge pull request #111663 from LanzaSchneider/patch-4

Fix compilation errors when `disable_xr=yes`

Repiteo committed:
Merge pull request #111672 from OverShifted/patch-1

Fix typo in mutex documentation

Repiteo committed:
Merge pull request #108862 from timothyqiu/plural-rules

Move context and plural support to `Translation`

Repiteo committed:
Merge pull request #107999 from timothyqiu/translation-cleanup

Clean up editor translation related methods

KnifeXRage committed:
Fix missing newline at end of ScrollContainer.xml

Add missing newline at the end of ScrollContainer.xml

KnifeXRage committed:
Fix missing newline at end of ScrollContainer.xml

Add missing newline at the end of ScrollContainer.xml

timothyqiu committed:
Fix error when compute shaders contain Unicode comment

KnifeXRage committed:
Fix XML formatting and improve descriptions in ScrollContainer

KnifeXRage committed:
Fix XML formatting and improve descriptions in ScrollContainer

KnifeXRage committed:
Change horizontal content align parameter type to work properly.

KnifeXRage committed:
Change horizontal content align parameter type to work properly.

KnifeXRage commented on scene/gui/scroll_container.cpp:
```diff
@@ -752,12 +820,29 @@ void ScrollContainer::_bind_methods() {
+
+	ClassDB::bind_integer_constant("ScrollContainer", "ContentVAlign", "V_ALIGN_TOP", V_ALIGN_TOP, false);
+	ClassDB::bind_integer_constant("ScrollContainer", "ContentVAlign", "V_ALIGN_CENTER", V_ALIGN_CENTER, false);
+	ClassDB::bind_integer_constant("ScrollContainer", "ContentVAlign", "V_ALIGN_BOTTOM", V_ALIGN_BOTTOM, false);
```
I tried it but it's not working. And engine compilation fails. But this way it works properly.

DarioSamo committed:
Do not begin a new frame during RenderingDevice's shutdown.

Repiteo committed:
Merge pull request #111523 from Joy-less/fix-IsStaticallyResolvable

Fix `IsStaticallyResolvable`

Repiteo committed:
Merge pull request #111658 from brycehutchings/bryceh_d3d12_native_handle_fixes

Fix D3D12 rendering device driver returning pointers to internal types for get_resource_native_handle instead of proper D3D12 primitives

Repiteo committed:
Merge pull request #110378 from timothyqiu/rtl-preview

Make text-related nodes translation domain aware

Repiteo committed:
Merge pull request #111211 from superherointj/enet-warning

System-provided ENet warning

Repiteo committed:
Merge pull request #111733 from DarioSamo/rendering-device-shutdown-fix

Do not begin a new frame during RenderingDevice's shutdown.

Repiteo committed:
Merge pull request #111509 from passivestar/orbit-snapping

Implement orbit snapping in 3D viewport

Repiteo committed:
Merge pull request #111703 from timothyqiu/glsl-utf8

Fix error when compute shaders contain Unicode comment

Repiteo committed:
Merge pull request #94047 from KoBeWi/resource_printer

Improve `to_string()` and add it to Resource

bruvzg committed:
[TextEdit] Make `wrap_right_offset` adjustable theme constant.

benbot committed:
fixes the lods array returned by mesh_get_surface

Ivorforce committed:
Fix false positive warning with `FixedVector` array bounds in gcc.

DallasHuff committed:
Moved File Filter When Filesystem Dock is on the Bottom Panel

When the Filesystem Dock is on the bottom panel vertical space is more
valuable than horizontal space. Moving the file filter to be horizontal
with the current path LineEdit makes the dock more space efficient when
on the bottom panel. When docked on the side, the file filter is
unchanged.

Repiteo committed:
Merge pull request #109792 from vaner-org/blendspace-better-cursor

Make BlendSpace selection tool consistent with StateMachine/BlendTree functionality

Repiteo committed:
Merge pull request #111745 from benbot/benbot/seems-wrong

Fix the LODs array returned by `mesh_get_surface`

Repiteo committed:
Merge pull request #111649 from KoBeWi/DESNAME

Improvements to ProjectManager's `_update_theme()`

Repiteo committed:
Merge pull request #109605 from vaner-org/statemachine-play-highlight-fix

Correctly reset StateMachine opacity when a node is played or renamed

Repiteo committed:
Merge pull request #109839 from vaner-org/blendspace-double-click

Double-click BlendSpace points to open their editor

Repiteo committed:
Merge pull request #109431 from dsnopek/openxr-composition-layer-safe-multithreaded-rendering

Make `OpenXRCompositionLayer` and its children safe for multithreaded rendering

Repiteo committed:
Merge pull request #111761 from Ivorforce/fixed-array-warn

Fix false positive warning with `FixedVector` array bounds in gcc.

Repiteo committed:
Core: Sidestep GCC false-positive

DeeJayLSP committed:
Editor: Add OpenType feature settings to Main Font

Kaleb-Reid committed:
Update note for rendering/rendering_device/vsync/swapchain_image_count

KnifeXRage committed:
Done Requested Changes

KnifeXRage committed:
Done Requested Changes

KnifeXRage committed:
Fixed Getter/Setter Methods

KnifeXRage committed:
Fixed Getter/Setter Methods

akien-mga committed:
Merge pull request #111771 from Repiteo/core/suppress-gcc-false-positive

Core: Sidestep GCC false-positive

KnifeXRage committed:
Fixed Docs

KnifeXRage committed:
Fixed Docs

AThousandShips commented on scene/gui/scroll_container.cpp:
```diff
@@ -370,18 +410,46 @@ void ScrollContainer::_reposition_children() {
+		{
+			float content_w = r.size.width;
+			float viewport_w_local = viewport_w;
+			int align_h = static_cast<int>(horizontal_content_align);
```
As is this

akien-mga committed:
Merge pull request #111299 from stuartcarnie/swift_build

Build: Fix container build path for swift

bruvzg committed:
[X11] Fix `keyboard_get_label_from_physical` errors when used on key without label.

stuartcarnie committed:
Build: Update to Xcode 26.0.1 for Apple builds

timothyqiu committed:
Fix crash when operating on newly created font RIDs

AThousandShips commented on scene/gui/scroll_container.cpp:
```diff
@@ -360,6 +360,46 @@ void ScrollContainer::_reposition_children() {
+		int align_v = static_cast<int>(vertical_content_align);
+		float extra = viewport_h - total_content_h;
+		if (align_v == CONTENT_V_ALIGN_CENTER) {
+			group_v_offset = floorf(extra * 0.5f);
```
```suggestion
			group_v_offset = Math::floor(extra * 0.5f);
```
Please use the built-in methods, for all these

AThousandShips commented on doc/classes/ScrollContainer.xml:
```diff
@@ -74,6 +77,9 @@
 			Overrides the [member ScrollBar.custom_step] used when clicking the internal scroll bar's vertical increment and decrement buttons or when using arrow keys when the [ScrollBar] is focused.
 		</member>
+		<member name="vertical_content_align" type="int" setter="set_vertical_content_align" getter="get_vertical_content_align" enum="ScrollContainer.ContentVAlign" default="0">
+			Controls Vertical Placement of Contents.
```
```suggestion
			Controls vertical placement of contents.
```

AThousandShips commented on doc/classes/ScrollContainer.xml:
```diff
@@ -108,6 +114,24 @@
 			Combines [constant SCROLL_MODE_AUTO] and [constant SCROLL_MODE_SHOW_ALWAYS]. The scrollbar is only visible if necessary, but the content size is adjusted as if it was always visible. It's useful for ensuring that content size stays the same regardless if the scrollbar is visible.
 		</constant>
+		<constant name="CONTENT_H_ALIGN_LEFT" value="0" enum="ContentHAlign">
+			Left horizontal alignment of contents. ([constant CONTENT_H_ALIGN_LEFT])
```
```suggestion
			Left horizontal alignment of contents.
```
For all

AThousandShips commented on scene/gui/scroll_container.cpp:
```diff
@@ -824,3 +909,29 @@ ScrollContainer::ScrollContainer() {
+	if (horizontal_content_align != new_align) {
+		horizontal_content_align = new_align;
+		queue_sort();
+	}
```
```suggestion
	ERR_FAIL_INDEX((int)p_align, 3);
	if (horizontal_content_align == p_align) {
		return;
	}
	horizontal_content_align = p_align;
	queue_sort();
```

Commit:
Correct Array `remove_at` class reference

`remove_at` can take negative positions as relative to the end of the array.
The note is no longer needed and contradictory to the first paragraph.

KoBeWi committed:
Add multilevel notes to methods

clayjohn committed:
Use GL_FRAMEBUFFER instead of GL_DRAW_FRAMEBUFFER when doing final blit to the screen framebuffer to work around OBS bug

blueskythlikesclouds committed:
Round values after renormalization when generating mipmaps.

akien-mga committed:
mbedTLS: Update to version 3.6.5

wjt committed:
TileSetAtlasSource: Fix and/an docstring typo

Repiteo committed:
Merge pull request #107671 from dsnopek/editor-run-control

Allow editor plugins to modify run arguments

Repiteo committed:
Merge pull request #111782 from Kaleb-Reid/update-swapchain-count-note

Update note for rendering/rendering_device/vsync/swapchain_image_count

Repiteo committed:
Merge pull request #111017 from jasonweiler/NewNodeDialogSplitter

Editor: Add splitter to "Create New Node" dialog

Repiteo committed:
Merge pull request #111661 from Repiteo/core/typed-container-migration-v2

Core: Consolidate typed container logic with `type_info.h`

Repiteo committed:
Merge pull request #111795 from bruvzg/x11_kbd_no_lbl

[X11] Fix `keyboard_get_label_from_physical` errors when used on key without label.

Repiteo committed:
Merge pull request #111156 from WhalesState/insp-groups

Group `virtual keyboard` and `word separators` properties in the inspector.

Repiteo committed:
Merge pull request #110131 from vaner-org/animationtree-better-deselection

Improve deselection in AnimationTree editors and inspector

Repiteo committed:
Merge pull request #111799 from stuartcarnie/update_xcode_version

Build: Update to Xcode 26.0.1 for Apple builds

Repiteo committed:
Merge pull request #111810 from timothyqiu/ts-null-data

Fix crash when operating on newly created font RIDs

Repiteo committed:
Merge pull request #108487 from raulsntos/dotnet/remove-efs-update

[.NET] Remove EFS update on reloading assemblies

Repiteo committed:
Merge pull request #111817 from charjr/fix-array-remove-at

Correct Array `remove_at()` class reference

Repiteo committed:
Merge pull request #108489 from raulsntos/dotnet/fix-bindgen-errors

[.NET] Fix various errors in bindings generator

Repiteo committed:
Merge pull request #111190 from SantiagoDaza3/Open-Source-Contribution

Fix 'LineEdit' Unicode code-point/control character insertion failing to emit text_changed

Repiteo committed:
Merge pull request #109534 from vaner-org/reconnect-statemachine-transitions

Allow reconnecting AnimationNodeStateMachine transitions

Repiteo committed:
Merge pull request #111841 from blueskythlikesclouds/mipmap-normalize-fix

Round values after renormalization when generating mipmaps.

Repiteo committed:
Merge pull request #111329 from clayjohn/GLES-disable-PVR-shader-cache

Add all PowerVR devices to the transform feedback shader cache ban list

Repiteo committed:
Merge pull request #111459 from paul-marechal/mesh_data_tool_tangent_doc

Add note for some usages of Plane in APIs

Repiteo committed:
Merge pull request #105450 from raulsntos/MissingNode/default_properties

Make `MissingNode`/`MissingResource` non-virtual and hide from dialogs

Repiteo committed:
Merge pull request #109043 from GabCoolDude/resource-signal-viewer

View resource signals in the Connections Dock

Repiteo committed:
Merge pull request #110709 from DeeJayLSP/dict-reserve

Add `reserve()` to `Dictionary`, apply to constructors on GDScript VM

kitbdev committed:
Fix NOTIFICATION_OS_IME_UPDATE docs on platforms

kitbdev committed:
Fix IME input in multiple Windows at once

eritschlSC committed:
Fix incorrect failure code in screen_get_framebuffer_format

RenderingDevice::screen_get_framebuffer_format should return a value of
type RenderingDevice::FramebufferFormatID which is an alias of int64_t
but it returns Error::FAILED with a value of 1. The compiler does not
complain because both types are integers but 1 corresponds to a valid
FramebufferFormatID, meaning a certain failure condition is missed.

This commit changes it to the correct value, INVALID_ID.

Repiteo committed:
Merge pull request #111744 from bruvzg/te_roff

[TextEdit] Make `wrap_right_offset` adjustable theme constant.

Repiteo committed:
Merge pull request #110989 from xuhuisheng/dev/snap

Fix snap Polygon2D editor

Repiteo committed:
Merge pull request #111378 from TokageItLab/loc-sp

Change Vector to LocalVector in SpringBoneSimulator3D

Repiteo committed:
Merge pull request #111773 from DeeJayLSP/main-font-calt-ot-features

Editor: Add OpenType feature settings to Main Font

Repiteo committed:
Merge pull request #111648 from kitbdev/fix-indented-line-wrap-nobreaks

Fix indented line wrap with no spaces

Repiteo committed:
Merge pull request #108320 from aaronfranke/gltf-buf-accessor-dict

GLTF: Move accessor and buffer view Dictionary conversion into those classes

Repiteo committed:
Merge pull request #109475 from nikitalita/preserve-gltf-names

Preserve mesh, material, and texture names in GLTF export

Repiteo committed:
Merge pull request #111296 from syntaxerror247/speed-control

Android Editor: Add game speed control options in game menu bar

Repiteo committed:
Merge pull request #111684 from ultsi/fix-lsp-non-existing-values-dict-access

LSP: Use safe methods to get data from dictionaries

Repiteo committed:
Merge pull request #111688 from Alex2782/android_export_webp

Fix custom icon in Android export

Repiteo committed:
Merge pull request #111691 from adamscott/fix-macos-microphone

[macOS] Fix microphone issue

Repiteo committed:
Merge pull request #111439 from bruvzg/fd_filt_add

Add MIME argument to the `FileDialog.add_filter`.

Repiteo committed:
Merge pull request #110937 from Calinou/editor-lightmapgi-bake-time-show-centiseconds

Show centiseconds in LightmapGI bake time printout in the editor

Repiteo committed:
Merge pull request #111456 from YeldhamDev/put_it_on_my_tab

Fix some dragging operations in the editor breaking when tabbing out

Repiteo committed:
Merge pull request #111852 from wjt/push-poqllplmlwyz

TileSetAtlasSource: Fix and/an docstring typo

Repiteo committed:
Merge pull request #111341 from aaronfranke/gltf-animate-visibility

GLTF: Support animating node visibility

Repiteo committed:
Merge pull request #111863 from kitbdev/fix-IME-UPDATE-description

Fix `NOTIFICATION_OS_IME_UPDATE` docs on platform availability

Repiteo committed:
Merge pull request #111865 from kitbdev/fix-multiple-ime

Fix IME input in multiple Windows at once

Repiteo committed:
Merge pull request #111615 from aaronfranke/gltf-anim-ptr-int-erpolate

GLTF: Enforce `STEP` interpolation for integer and boolean animations

DeeJayLSP committed:
Use Inter as the default editor font, features enabled

Repiteo committed:
Merge pull request #102020 from ryevdokimov/move-viewport-gizmos-visibility-out-of-process

Move editor viewport gizmos visibility logic out of process notification

Repiteo committed:
Merge pull request #102021 from ryevdokimov/remove-redunant-previewing-cinema-line

Remove redundant line when toggling cinematic preview

Repiteo committed:
Merge pull request #110855 from arkology/project-manager-favorites

`ProjectManager`: Change favorite button tooltip to toggle

Repiteo committed:
Merge pull request #111883 from eritschlSC/fix-framebuffer-format

Fix incorrect failure code in screen_get_framebuffer_format

Repiteo committed:
Merge pull request #110871 from KoBeWi/fold_code

Reduce repetitive code in FoldableContainer

Repiteo committed:
Merge pull request #111000 from DallasHuff/move-file-filter

Move file filter when filesystem dock is in the bottom panel

Repiteo committed:
Merge pull request #109725 from BlueCube3310/betsy-rt-bc1-cache

Betsy: Generate BC1 table cache during compression

Repiteo committed:
Merge pull request #111135 from m4gr3d/add_emit_signal_overload

[Android] Minor updates to the `GodotPlugin` APIs

Repiteo committed:
Merge pull request #111015 from Alexofp/texture-compression-build-option

Scons option to enable CVTT and Betsy compression in export templates

Repiteo committed:
Merge pull request #102055 from ryevdokimov/condition-editor-viewport-shortcuts

Condition editor viewport nav modifier checks on freelook / mouse is captured

Repiteo committed:
Merge pull request #110641 from DallasHuff/new-resource-hotkey

Add Create Resource Hotkey

Repiteo committed:
Merge pull request #110647 from arkology/add-meta-dialog

Expand `LineEdit` with metadata name in `AddMetadataDialog`

Repiteo committed:
Merge pull request #110904 from YeldhamDev/tree_drag_unfolding

Unfold tree items on hover while drag-n-dropping

Repiteo committed:
Merge pull request #110910 from YeldhamDev/texture_region_drag_cancel

Fix "Region Editor" stuck in drag state when closing

Repiteo committed:
Merge pull request #106947 from Meorge/feat/quick-load-preview

Allow Quick Open dialog to preview change in scene

Repiteo committed:
Merge pull request #110942 from KoBeWi/the_what_array

Fix Script editor state types

Repiteo committed:
Merge pull request #101092 from HolonProduction/argument-options-completion-type

Autocompletion: Use correct completion type for argument options

Repiteo committed:
Merge pull request #111210 from blueskythlikesclouds/ssr-overhaul

Overhaul screen space reflections.

Repiteo committed:
Merge pull request #110060 from BlueCube3310/betsy-rgb

Betsy: Convert RGB to RGBA on the GPU for faster compression

Repiteo committed:
Merge pull request #111478 from HolonProduction/lsp-decl-fix

LSP: Fix goto native declaration

Repiteo committed:
Merge pull request #108534 from Giganzo/breeze-theme-color-update

Update Breeze dark theme colors

Repiteo committed:
Merge pull request #110972 from Tcarr20/fix-editor-log-bbcode-search

Fix editor log search ignoring BBCode formatting in messages

Repiteo committed:
Merge pull request #111140 from DeeJayLSP/gdinter

Use Inter as the default editor font, features enabled

Repiteo committed:
Merge pull request #110895 from Giganzo/input-focus

Add setting for when to show the focus state for mouse input

Repiteo committed:
Merge pull request #110127 from AayushSarikhada/android-back-gesture-fail-fix

Fix Android back gesture failing after keyboard dismissal

Repiteo committed:
Merge pull request #109971 from clayjohn/hdr2d-mobile

Use half float precision buffer for 3D when HDR2D is enabled

Repiteo committed:
Merge pull request #103478 from KoBeWi/hover_witch

Add switch on hover to TabBar

Repiteo committed:
Merge pull request #94488 from Alex2782/fix_create_folder_94446

Fix `exclusive` child window

Repiteo committed:
Merge pull request #111834 from clayjohn/OBS-bug

Use `GL_FRAMEBUFFER` instead of `GL_DRAW_FRAMEBUFFER` when doing final blit to the screen framebuffer to work around OBS bug

Repiteo committed:
Merge pull request #109596 from atlasapplications/splash-screen-stretch-mode

Add Stretch Modes for Splash Screen

Repiteo committed:
Merge pull request #107487 from roughbits01/fix-shortcut-names

Remove the need to have a display name for built-in action overrides

kleonc committed:
Fix Control.pivot_offset missing PROPERTY_USAGE_STORAGE flag

Repiteo committed:
Merge pull request #110433 from devloglogan/custom-monitor-types

Allow custom debug monitors to select desired type

Repiteo committed:
Merge pull request #85762 from KoBeWi/wait_it's_all_nodes︖

Add `change_scene_to_node()`

Repiteo committed:
Merge pull request #107843 from DexterFstone/fix/tilemap-tool-override

Fix: Tilemap tools not overriding main editor tools

Repiteo committed:
Merge pull request #109220 from precup/avoid-spurious-curve-changed-signals

Remove spurious `Curve3D` `changed` signals

Repiteo committed:
Merge pull request #110308 from BlueCube3310/image-mip-renorm-fix

Image: Fix normalization of mipmaps for half and float formats

Repiteo committed:
Merge pull request #107303 from passivestar/close-dialog-action

Add support for closing dialog windows with Cmd+W on macOS

Repiteo committed:
Merge pull request #111847 from akien-mga/akien/embedded-run-respect-override

EditorRun: Load `override.cfg` to get window configuration for embedded mode

Repiteo committed:
Merge pull request #111437 from Kaleb-Reid/compat-fix-proximity-fade

Use correct ndc for proximity fade in Compatibility

Repiteo committed:
Merge pull request #111921 from kleonc/control_pivot_offset_fix_missing_storage_flag

Fix `Control.pivot_offset` missing `PROPERTY_USAGE_STORAGE` flag

Repiteo committed:
Merge pull request #111830 from KoBeWi/documentation_on_multiple_levels

Add multilevel notes to methods

Repiteo committed:
Merge pull request #109753 from dsnopek/openxr-ext-threading-docs

`OpenXRExtensionWrapper`: Document how to safely work when rendering with "Separate" thread model

Repiteo committed:
Merge pull request #111162 from KoBeWi/tfw_your_side_projects_goes_to_trash

Remove side menu functionality in `EditorFileDialog`

Repiteo committed:
Merge pull request #111325 from TokageItLab/state-default-st

Make StateMachinePlayback set `Start` state as default in constructor

Repiteo committed:
Merge pull request #110271 from BlueCube3310/image-conv-opti-incompatible

Image: Optimize manual format conversion

akien-mga committed:
Merge pull request #111845 from akien-mga/mbedtls-3.6.5

mbedTLS: Update to version 3.6.5

KnifeXRage committed:
Merge branch 'content_alignment_scroll_container' of https://github.com/KnifeXRage/godot-fork into content_alignment_scroll_container

KnifeXRage committed:
Revert "Fixed Docs"

This reverts commit 8c1a1310c0223f4e46589291b9cda3873500960c.
